dev-packages/application-manager/src/application-package-manager.ts

```diff
@@ -78,26 +78,41 @@ export class ApplicationPackageManager {
     }
 
     async startElectron(args: string[]): Promise<void> {
+        const { mainArgs, options } = this.adjustArgs([this.pck.frontend('electron-main.js'), ...args]);
         const electronCli = require.resolve('electron/cli.js', { paths: [this.pck.projectPath] });
-        this.process.spawn(electronCli, [this.pck.frontend('electron-main.js'), ...args],
-            { stdio: [0, 1, 2] });
+        this.__process.fork(electronCli, mainArgs, options);
     }
 
     async startBrowser(args: string[]): Promise<void> {
-        const options: cp.ForkOptions = {
-            stdio: [0, 1, 2, 'ipc'],
-            env: {
-                ...process.env,
-                THEIA_PARENT_PID: String(process.pid)
-            }
+        const { mainArgs, options } = this.adjustArgs(args);
+        this.__process.fork(this.pck.backend('main.js'), mainArgs, options);
+    }
+
+    private adjustArgs(args: string[], forkOptions: cp.ForkOptions = {}): Readonly<{ mainArgs: string[]; options: cp.ForkOptions }> {
+        const options = {
+            ...this.forkOptions,
+            forkOptions
         };
         const mainArgs = [...args];
         const inspectIndex = mainArgs.findIndex(v => v.startsWith('--inspect'));
         if (inspectIndex !== -1) {
             const inspectArg = mainArgs.splice(inspectIndex, 1)[0];
             options.execArgv = ['--nolazy', inspectArg];
         }
-        this.__process.fork(this.pck.backend('main.js'), mainArgs, options);
+        return {
+            mainArgs,
+            options
+        };
+    }
+
+    private get forkOptions(): cp.ForkOptions {
+        return {
+            stdio: [0, 1, 2, 'ipc'],
+            env: {
+                ...process.env,
+                THEIA_PARENT_PID: String(process.pid)
```

Comment:
Why do we keep setting this variable instead of using [`process.ppid`](https://nodejs.org/docs/latest-v10.x/api/process.html#process_process_ppid) ?

Reply:
I did not want to change the original logic:
https://github.com/theia-ide/theia/blob/dd9eb6b62419187dbd2616bb747fd7b321502019/dev-packages/application-manager/src/application-package-manager.ts#L91

Reply:
That I understand, we should not in this PR, but I was just wondering if you knew why it was like this in the first place. It is not really important for this PR :)